api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -205,38 +208,74 @@ public String toString() {
   }
 
   public static class TimestampType extends PrimitiveType {
-    private static final TimestampType INSTANCE_WITH_ZONE = new TimestampType(true);
-    private static final TimestampType INSTANCE_WITHOUT_ZONE = new TimestampType(false);
 
+    private static final TimestampType INSTANCE_MICROS_WITH_ZONE =
+        new TimestampType(true, ChronoUnit.MICROS);
+    private static final TimestampType INSTANCE_MICROS_WITHOUT_ZONE =
+        new TimestampType(false, ChronoUnit.MICROS);
+    private static final TimestampType INSTANCE_NANOS_WITH_ZONE =
+        new TimestampType(true, ChronoUnit.NANOS);
+    private static final TimestampType INSTANCE_NANOS_WITHOUT_ZONE =
+        new TimestampType(false, ChronoUnit.NANOS);
+
+    /** @deprecated use {@link #microsWithZone()} instead. */
+    @Deprecated
```

Comment:
Can you add when this will be removed? I think that would be in the 2.0 release.

Reply:
Happy to add that. Are we certain it will be in "2.0" or should I say something like "likely removed in 2.0"?

Reply:
Be specific. We can remove it in 2.0 so we should.

Reply:
Done.